### PR TITLE
Fix bug for non-unique initial IndexMap.update

### PR DESCRIPTION
**Pull request by collijk**
### Description

*Category*: bugfix
*JIRA issue*: 

- Added regression testing for bug where IndexMap.update logic would not treat duplicate initial mapping data as duplicates.
- Fixed bug.
- Removed `"entrance_time"` as a default key_column for the randomness system as it's not generated by anything in base `vivarium`
- Added `"entrance_time"` and `"age"` to the `base_config` test fixture so the `TestPopulation` component will work.

### Testing
New tests plus normal CI

### CODE REVIEW

File: tests/framework/randomness/test_index_map.py [hunk 1]
```diff
@@ -164,31 +164,49 @@ def test_hash_uniformity(map_size_and_hashed_values):
...
+
+    index_map.update(keys)
+    with pytest.raises(RandomnessError):
+        index_map.update(keys)
+
+
+def test_update_empty_good_keys(index_map):
+    keys = generate_keys(1000)
+    index_map.update(keys)
+    assert len(index_map._map) == len(keys), "All keys not in mapping"
```
Comment (rmudambi) [inline tests/framework/randomness/test_index_map.py:198]: Should have different error messages for these two error cases. 

Minor point, but I think any two of these asserts is sufficient to test all keys are present and are unique.

File: tests/framework/randomness/test_index_map.py [hunk 2]
```diff
@@ -164,31 +164,49 @@ def test_hash_uniformity(map_size_and_hashed_values):
...
+
+
+def test_update_nonempty_good_keys(index_map):
+    keys = generate_keys(2000)
+    keys1, keys2 = keys[:1000], keys[1000:]
+
+    index_map.update(keys1)
+    index_map.update(keys2)
+
+    assert len(index_map._map) == len(keys), "All keys not in mapping"
```
Comment (rmudambi) [inline tests/framework/randomness/test_index_map.py:210]: Same as above.

**Review by rmudambi — APPROVED** [review]

File: src/vivarium/framework/randomness/index_map.py [hunk 3]
```diff
@@ -39,8 +39,10 @@ def update(self, new_keys: pd.Index) -> None:
         """
         if new_keys.empty or not self._use_crn:
             return  # Nothing to do
-        elif not self._map.index.intersection(new_keys).empty:
```
[the remark this answers is not in the document]
Comment (collijk) [reply src/vivarium/framework/randomness/index_map.py:42]: It did not handle duplicates within the new keys only values duplicated between the existing map and the new keys.

File: tests/framework/randomness/test_index_map.py (hunk 1 above)
Comment (collijk) [reply tests/framework/randomness/test_index_map.py:198]: Fixed in next pr

**Pull request merged**